lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVMBase.h

```diff
@@ -305,9 +306,10 @@ class ConvertTritonGPUOpToLLVMPatternBase {
       // extract multi dimensional index for current element
       auto idx = srcIndices[elemIdx];
       Value idxCol = idx[outOrder[0]]; // contiguous dimension
-      Value idxRow = idx[outOrder[1]]; // discontiguous dimension
+      Value idxRow =
+          rank == 2 ? idx[outOrder[1]] : i32_val(0); // discontiguous dimension
```

Comment:
Might have to separate code paths for 1D tensors if this causes a perf regression